### DISCUSSION TITLE
Ghostty mod key events register on all terminals.

### DISCUSSION THREAD

**Discussion by ngynkvn** [ghostty-org/ghostty#2708] [Q&A] [answered]
I’m not sure if this warrants a GitHub issue, but I noticed that Ghostty sends modifier key events to all running windows and panes, not just the active pane. I’ve been wondering if this is intentional or not. I have a video clip if you’d like to see proof.

I ran a comparison test in Wezterm, and it didn’t exhibit this behavior. I can try other applications as well. (It doesn’t duplicate MOD+Key events, only MOD events, such as CMD, CMD+ALT, etc.)

I’ve only observed this behavior on macOS so far, but I can check on Linux and Windows if desired.

Testing for this should be straightforward. Open a window, split it into two panes, and then open a terminal inspector in both panes. Check the Keyboard tab to see if you can reproduce the issue. 

Since this doesn’t affect “proper chords” like CMD+ALT+A, I don’t anticipate it impacting the majority of users.

**Comment by mitchellh** [comment, marked as answer]
It is intended that at the apprt level we receive the modifier event but not intended that we encode it. If it is being encoded for the pty it's a bug. The reason we want to receive the modifier is so that URLs in non-focused surfaces can be highlighted or unhighlighted (there was another issue we resolved related to this).

**Reply by ngynkvn** [reply]
<details>
<summary>Previous Reply</summary>

If I understand this correctly, the events are indeed being sent to the pty. I noticed this first from `tail`ing a log file as I was developing something with libvaxis. i.e 
![image](https://github.com/user-attachments/assets/286cd273-00f8-41e4-9e88-6dc7a50bc849)

Those last few logs were from me pressing "CMD" and "Shift" inside the nvim editor

Is it correct to say that the top right panel is a "pty" or am I misunderstanding something?

</details>

Did some more research and in that case I don't think this would be considered a bug

**Reply by mitchellh** [reply]
> Did some more research and in that case I don't think this would be considered a bug

Can you expand on this? I think it may be...
